### PR TITLE
`Plagiarism checks`: Highlight relevant files when comparing programming exercise submissions

### LINKED ISSUE #5066: `Plagiarism Checks`: Highlight changed/non template files in file comparison view of programming exercises
### Is your feature request related to a problem?

When comparing suspected plagiarism cases of programming exercises, there are often files that are part of the template and not changed and therefore don't contain any similar elements. For extensive exercises, it is therefore necessary to go through a number of files that aren't relevant to the actual plagiarism case which can be very time-consuming.

E.g last weeks EIST Homework contained a lot of files that weren't touched by students and have to all be checked

image

### Describe the solution you'd like

Highlight the files that actually contain any similar elements that are relevant for plagiarism checks, e.g., by increasing the font-weight of the file name
image

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #5141: Text in dropdown in plagiarism check view is not visible in dark mode
### Describe the bug

The text in the file dropdown is not visible in the dark mode.

### To Reproduce

1. Go to a text/programming exercise with plagiarism cases.
2. Enable dark mode.
3. Select one plagiarism case.
4. Check that the text inside the dropdown is not visible

### Expected behavior

The text in the dropdown should have non-white (e.g.) black color.

### Screenshots

Bildschirmfoto 2022-05-23 um 12 47 47

Bildschirmfoto 2022-05-23 um 12 47 52

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- ~[ ] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).~
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- ~[ ] I translated all newly inserted strings into English and German.~

### Motivation and Context
Closes #5066 

### Description
The file list as presented had two main issues:
- The files were listed in a random order, different for both students.
- The files for which actual plagiarism matches have been found were not highlighted.

Both issues make it hard to see which files need to be compared manually when checking for plagiarism. Therefore, this PR
- Highlights files with plagiarism matches in bold text.
- Sorts all files in a fixed order so that files with matches are grouped together at the top of the list. With the groups of files with/without matches the files are then sorted lexicographically.

Additional fixes:
- Use a proper dropdown element for the file selection.
- Limit the vertical size of the dropdown list and add scrolling so that it no longer gets cut off when there are many files. (E.g., in the before screenshot below there is actually at least one more file.)
- Closes #5141: Dropdown text not readable in dark mode, student names in sidebar hardly readable in dark mode

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Programming exercise that has multiple student submissions so that a plagiarism report can be created.
- 1 Text exercise that has multiple student submissions so that a plagiarism report can be created.

1. Go to the plagiarism page for the programming exercise (white button ‘Check Plagiarism’ in the button group at the top of the exercise page).
2. Create a plagiarism report for this exercise.
    - Click on the ‘Plagiarism Detection’ header to open the options.
    - Tick ‘Generate JPlag Report’.
    - Generate the plagiarism report.
4. Select a student pair and check that the dropdown list of files is ordered as described above.
5. (optional) The JPlag-Report is a downloaded zip-File that contains a HTML document. In there you can select the same student pair and verify that all files listed there are highlighted in the dropdown in Artemis.
6. Open the plagiarism page for a text exercise. It should not have changed and still work without errors.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| `text-submission-viewer` | 50% | 75% |
| `split-pane-header` | 91% | 100% |

### Screenshots
Before:
Screenshot 2022-05-14 at 06-12-36 Plagiarism Detection

After:
The bold item in the list is just coincidentally the currently selected file, as in this case there is only one potential plagiarism case in this file. Multiple files would be marked bold otherwise.
Screenshot 2022-05-23 at 14-52-44 Plagiarism Detection
Screenshot 2022-05-23 at 14-52-15 Plagiarism Detection

### CODE REVIEW

File: src/test/javascript/spec/component/plagiarism/text-submission-viewer.component.spec.ts [hunk 1]
```diff
@@ -75,10 +75,46 @@ describe('Text Submission Viewer Component', () => {
         expect(comp.isProgrammingExercise).toBe(true);
     });
 
+    it('sort and filter the files when fetching a programming submission', () => {
```
Comment (b-fein) [inline src/test/javascript/spec/component/plagiarism/text-submission-viewer.component.spec.ts:78]:
```suggestion
    it('sorts and filters the files when fetching a programming submission', () => {
```

**Review by rhubinger — COMMENTED** [review]
While testing I couldn't open any files from the left student. The right student's files seemed to work but I couldn't compare anything because the left student's files weren't visible. Screenshot (14)

**Review by Lugitan — APPROVED** [review]
Tested on TS1 in testin session. Works as expected.

**Review by canberkanar — DISMISSED** [review]
Tested on ts1 in the testing session in accordance with the instructions. All functionality works correctly and plagiarism detection for text exercises is still fine too.

**Comment by manuelmanso** [comment]
Tested during testing session on TS1. Works as expected 👍

**Review by ole-ve — DISMISSED** [review]
Tested on TS1 during testing session, worked fine for me 👍

**Review by akesfeden — APPROVED** [review]
Tested on TS1 during testing session, works as described, nice improvements!

**Review by manuelmanso — APPROVED** [review]
Tested during testing session on TS1. Works as expected 👍

**Review by minrows — APPROVED** [review]
Tested in the testing session on TS1, works as described.

**Review by ole-ve — APPROVED** [review]
I just checked the coloring on TS1, looks good to me.
Nothing else changes, reapprove.

**Review by babyygemperor — APPROVED** [review]
Tested on ts1, works as described

**Comment by Dominik-Weinzierl** [comment]
I will review it later this evening! :D I have already some minor suggestions :)

File: src/test/javascript/spec/component/plagiarism/text-submission-viewer.component.spec.ts [hunk 2]
```diff
@@ -75,10 +75,46 @@ describe('Text Submission Viewer Component', () => {
...
+        comp.matches.set('kContinuedName', [{ from: new TextSubmissionElement(), to: new TextSubmissionElement() }]);
+
+        jest.spyOn(repositoryService, 'getRepositoryContent').mockReturnValue(of(filesUnordered));
+
+        comp.ngOnChanges({
+            plagiarismSubmission: { currentValue: { submissionId: 2 } } as SimpleChange,
+        });
+
+        expect(repositoryService.getRepositoryContent).toHaveBeenCalled();
+        expect(comp.isProgrammingExercise).toBe(true);
```
Comment (Dominik-Weinzierl) [inline src/test/javascript/spec/component/plagiarism/text-submission-viewer.component.spec.ts:101]:
```suggestion
        expect(comp.isProgrammingExercise).toBeTrue();
```

**Review by Dominik-Weinzierl — COMMENTED** [review]
Code looks good to me :+1: some minor remarks but i was really nit-picky :D

**Review by Dominik-Weinzierl — APPROVED** [review]
Code looks good to me :D thank you for implementing my suggestions!

**Review by JohannesStoehr — APPROVED** [review]
Code looks good

**Pull request merged**